Source/DafnyCore/Generic/Node.cs

```diff
@@ -10,9 +10,15 @@
 namespace Microsoft.Dafny;
 
 public interface INode {
+  public IToken StartToken => RangeToken.StartToken;
+  public IToken EndToken => RangeToken.EndToken;
+  IEnumerable<IToken> OwnedTokens { get; }
   RangeToken RangeToken { get; }
   IToken Tok { get; }
-  IEnumerable<Node> Children { get; }
+  IEnumerable<INode> Children { get; }
+  IEnumerable<INode> PreResolveChildren { get; }
+
+  IEnumerable<INode> GetConcreteChildren();
```

Comment:
What's the difference between this and Children and PreResolveChildren? A comment would be appreciated.